modules/core/shared/src/main/scala/Trace.scala

```diff
@@ -49,6 +49,11 @@ trait Trace[F[_]] {
     */
   def traceId: F[Option[String]]
 
+  /** A unique ID for this span, if available. This can be useful to include in error messages for
+    * example, so you can quickly find the associated trace.
+    */
+  def spanId: F[Option[String]]
```

Comment:
If we want to preserve backwards compatibility we'll have to do something like this so that there is a default implementation. Otherwise any implementations outside of this repo will break.

```suggestion
  /** A unique ID for this span, if available. This can be useful to include in error messages for
    * example, so you can quickly find the associated trace.
    */
  def spanId(implicit F: Applicative[F]): F[Option[String]] = F.pure(None)
```

Reply:
No, I don't think so. Anyone using `Tracer[F]` where there is no `Applicative[F]` would simply be unable to call this method.

Reply:
Oh right, because the constraint is on the method 😅. Makes sense to me!